capellambse/svg/drawing.py

```diff
@@ -911,57 +906,36 @@ def render_hbounded_lines(
     )
 
 
-def get_label_position_x(
-    builder: LabelBuilder, lines: LinesData, render_icon: bool
+def get_label_position(
+    builder: LabelBuilder, lines: LinesData
 ) -> tuple[float, float]:
-    """Return x-coordinate of label-text and icon."""
-    icon_size = (builder.icon_size + decorations.icon_padding) * render_icon
-    sh_icon_size = (
-        builder.icon_size + decorations.icon_padding / 2
-    ) * render_icon
+    """Return the initial label position (x, y)."""
+    icon_size = (builder.icon_size + decorations.icon_padding) * builder.icon
     if builder.text_anchor == "start":
-        x = builder.label["x"] + lines.margin + icon_size
-        return x, x - sh_icon_size
-    x = builder.label["x"] + (builder.label["width"] + icon_size) / 2
-    return x, x - (lines.max_line_width / 2 + sh_icon_size)
-
+        builder.label["x"] += lines.margin + icon_size
+    else:
+        builder.label["x"] += (builder.label["width"] + icon_size) / 2
 
-def get_label_position_y(builder: LabelBuilder, lines: LinesData) -> float:
-    """Return y-coordinate of label-text."""
     dominant_baseline_adjust = chelpers.extent_func(lines.lines[0])[1] / 2
     if builder.y_margin is None:
         builder.y_margin = (builder.label["height"] - lines.text_height) / 2
-    return builder.label["y"] + builder.y_margin + dominant_baseline_adjust
+    builder.label["y"] += builder.y_margin + dominant_baseline_adjust
+    return builder.label["x"], builder.label["y"]
 
 
 def get_label_icon_position(
-    builder: LabelBuilder, text_height: int | float, icon_x: int | float
+    builder: LabelBuilder, lines: LinesData
 ) -> tuple[float, float]:
-    """Calculate the position of the icon relative to the label.
-
-    Parameters
-    ----------
-    builder : LabelBuilder
-        LabelBuilder
-    text_height : int | float
-        The height of the text in the label.
-    icon_x : int | float
-        The x position of the icon.
-
-    Returns
-    -------
-    icon_coords : tuple[float, float]
-        The x and y coordinates of the icon.
-    """
-    assert builder.y_margin is not None
-    icon_y = (
-        builder.label["y"]
-        + builder.y_margin
-        + (text_height - builder.icon_size) / 2
+    """Calculate the icon's position."""
+    icon_y = builder.label["y"] - builder.icon_size / 2
+    if lines.text_height > lines.line_height:
+        icon_y += lines.text_height / 2 - decorations.icon_padding * 2
+        if builder.label["class"] == "Annotation":
+            icon_y -= decorations.icon_padding
+
+    icon_x = (
+        builder.label["x"] - builder.icon_size - decorations.icon_padding * 2
     )
-    if (
-        icon_x < builder.label["x"] - decorations.icon_padding
-        and builder.label["class"] != "Annotation"
-    ):
-        icon_x = builder.label["x"] - decorations.icon_padding
-    return icon_x, icon_y
+    if builder.text_anchor == "middle":
+        icon_x -= lines.max_line_width / 3
```

Comment:
Please don't ask why it is `/3` mathematically this doesn't make any sense. It should be `/2` (half) but,... the physics in this game are sth. else.